src/transaction_builder.js

```diff
@@ -44,87 +44,102 @@ function fixMSSignatures (transaction, vin, pubKeys, signatures, prevOutScript,
+        result.redeemScript = redeemScript
+        result.redeemScriptType = redeemScriptType
 
-      break
+        return result
 
-    case 'pubkey':
-      parsed = ECSignature.parseScriptSignature(scriptSigChunks[0])
-      hashType = parsed.hashType
-      signatures = [parsed.signature]
+      case 'pubkeyhash':
+        parsed = ECSignature.parseScriptSignature(scriptSigChunks[0])
+        hashType = parsed.hashType
+        pubKeys = scriptSigChunks.slice(1)
+        signatures = [parsed.signature]
+        prevOutScript = bscript.pubKeyHashOutput(bcrypto.hash160(pubKeys[0]))
 
-      if (redeemScript) {
-        pubKeys = redeemScriptChunks.slice(0, 1)
-      }
+        break
+
+      case 'pubkey':
+        parsed = ECSignature.parseScriptSignature(scriptSigChunks[0])
+        hashType = parsed.hashType
+        signatures = [parsed.signature]
+
+        if (redeemScriptChunks) {
+          pubKeys = redeemScriptChunks.slice(0, 1)
+        }
 
-      break
+        break
 
-    case 'multisig':
-      signatures = scriptSigChunks.slice(1).map(function (chunk) {
-        if (chunk === ops.OP_0) return undefined
+      case 'multisig':
+        signatures = scriptSigChunks.slice(1).map(function (chunk) {
+          if (chunk === ops.OP_0) return undefined
 
-        var parsed = ECSignature.parseScriptSignature(chunk)
-        hashType = parsed.hashType
+          parsed = ECSignature.parseScriptSignature(chunk)
+          hashType = parsed.hashType
 
-        return parsed.signature
-      })
+          return parsed.signature
+        })
 
-      if (redeemScript) {
-        pubKeys = redeemScriptChunks.slice(1, -2)
+        if (redeemScriptChunks) {
+          pubKeys = redeemScriptChunks.slice(1, -2)
 
-        if (pubKeys.length !== signatures.length) {
-          signatures = fixMSSignatures(transaction, vin, pubKeys, signatures, redeemScript, hashType, redeemScript)
+          if (pubKeys.length !== signatures.length) {
+            signatures = fixMSSignatures(transaction, vin, pubKeys, signatures, bscript.compile(redeemScriptChunks), hashType, redeemScript)
+          }
         }
-      }
 
-      break
+        break
+    }
+
+    return {
+      hashType: hashType,
+      pubKeys: pubKeys,
+      signatures: signatures,
+      prevOutScript: prevOutScript,
+      redeemScript: redeemScript,
+      redeemScriptType: redeemScriptType
+    }
   }
 
-  return {
-    hashType: hashType,
-    prevOutScript: prevOutScript,
+  // Extract hashType, pubKeys, signatures and prevOutScript
+  var result = processScript(prevOutType, scriptSigChunks)
+
+  var finalresult = {
+    hashType: result.hashType,
+    prevOutScript: result.prevOutScript,
     prevOutType: prevOutType,
-    pubKeys: pubKeys,
-    redeemScript: redeemScript,
-    scriptType: scriptType,
-    signatures: signatures
+    pubKeys: result.pubKeys,
+    redeemScript: result.redeemScript,
+    redeemScriptType: result.redeemScriptType,
```

Comment:
renamed `scriptType` to `redeemScriptType` as preperation for segwit where we'll also have `segWitScriptType`